client/a8c-for-agencies/components/add-new-site-button/import-from-wpcom-modal/wpcom-sites-table.tsx

```diff
@@ -0,0 +1,190 @@
+import { WordPressLogo, JetpackLogo, Gridicon } from '@automattic/components';
+import { useDesktopBreakpoint } from '@automattic/viewport-react';
+import { CheckboxControl } from '@wordpress/components';
+import { Icon } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { useState, useMemo, useCallback } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import TextPlaceholder from 'calypso/a8c-for-agencies/components/text-placeholder';
+import useFetchDashboardSites from 'calypso/data/agency-dashboard/use-fetch-dashboard-sites';
+import { urlToSlug } from 'calypso/lib/url/http-utils';
+import { useSelector } from 'calypso/state';
+import { getActiveAgencyId } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import getSites from 'calypso/state/selectors/get-sites';
+import getIsSiteWPCOM from 'calypso/state/selectors/is-site-wpcom';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import type { Site } from 'calypso/a8c-for-agencies/sections/sites/types';
+
+type SiteItem = {
+	id: number;
+	site: string;
+	date: string;
+	type: string;
+};
+
+const TypeIcon = ( { siteId }: { siteId: number } ) => {
+	const isWPCOM = useSelector( ( state ) => getIsSiteWPCOM( state, siteId ) );
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+
+	if ( isWPCOM ) {
+		return <Icon className="wpcom-sites-table__icon" icon={ <WordPressLogo /> } />;
+	}
+	if ( isJetpack ) {
+		return <Icon className="wpcom-sites-table__icon" icon={ <JetpackLogo /> } />;
+	}
+	return <Gridicon icon="minus" />;
+};
+
+export default function WPCOMSitesTable( {
+	selectedSites,
+	setSelectedSites,
+}: {
+	selectedSites: number[];
+	setSelectedSites: ( sites: number[] ) => void;
+} ) {
+	const translate = useTranslate();
+	const agencyId = useSelector( getActiveAgencyId );
+
+	const { data, isFetching } = useFetchDashboardSites( {
+		isPartnerOAuthTokenLoaded: false,
+		searchQuery: '',
+		currentPage: 1,
+		sort: {
+			field: '',
+			direction: '',
+		},
+		perPage: 100,
+		agencyId,
+		filter: {
+			issueTypes: [],
+			showOnlyFavorites: false,
+		},
+	} );
```

Comment:
As I mentioned, are we planning to fetch all Agency sites instead of only the first 100?  I'm okay doing this on a separate PR.

Reply:
I think this should be done in a different PR. I have created a ticket: https://github.com/Automattic/automattic-for-agencies-dev/issues/762